src/utils/CometAdapter.cpp

```diff
@@ -0,0 +1,589 @@
+    registerIntOption_("use_Z_ions", "<num>", 0, "use Z ions for PSM, 0 == no, 1 == yes", false, true);
+    registerIntOption_("use_NL_ions", "<num>", 0, "use Neutral Loss ions for PSM, 0 == no, 1 == yes", false, true);
+
+    //Output
+    registerIntOption_("num_hits", "<num>", 5, "Number of peptide hits in output file", false, false);
+
+    //mzXML/mzML parameters
+    registerStringOption_("precursor_charge", "[min] [max]", "0 0", "charge range to search: 0 0 == search all charges, 2 6 == from +2 to +6, 3 3 == +3", false, false);
+    registerIntOption_("override_charge", "<num>", 0, "0 = keep any known precursor charge state, 1 = ignore known precursor charge state and use precursor_charge parameter, 2=ignore precursor charges outside precursor_charge range, 3= keep any known precursor charge state. For unknown charge states, search as singly charged if there is no signal above the precursor m/z or use the precursor_charge range", false, false);
+    registerIntOption_("ms_level", "<num>", 2, "MS level to analyze, valid are levels 2 (default) or 3", false, false);
+    registerStringOption_("activation_method", "<method>", "ALL", "activation method; used if activation method set; allowed ALL, CID, ECD, ETD, PQD, HCD, IRMPD", false, false);
+    setValidStrings_("activation_method", ListUtils::create<String>("ALL,CID,ECD,ETD,PQD,HCD,IRMPD"));
+
+    //Misc. parameters
+    registerStringOption_("digest_mass_range", "[min] [max]", "600.0 5000.0", "MH+ peptide mass range to analyze", false, true);
+    registerIntOption_("max_fragment_charge", "<num>", 3, "set maximum fragment charge state to analyze (allowed max 5)", false, false);
+    registerStringOption_("max_precursor_charge", "<num>", "0+", "set maximum precursor charge state to analyze (allowed max 9)", false, true);
+    registerIntOption_("clip_nterm_methionine", "<num>", 0, "0=leave sequences as-is; 1=also consider sequence w/o N-term methionine", false, false);
+    registerIntOption_("spectrum_batch_size", "<num>", 0, "max. // of spectra to search at a time; 0 to search the entire scan range in one loop", false, true);
+    registerDoubleOption_("mass_offsets", "<offset>", 0, "one or more mass offsets to search (values substracted from deconvoluted precursor mass)", false, true);
+
+    // spectral processing
+    registerIntOption_("minimum_peaks", "<num>", 10, "required minimum number of peaks in spectrum to search (default 10)", false, true);
+    registerIntOption_("minimum_intensity", "<num>", 0, "minimum intensity value to read in", false, true);
+    registerIntOption_("remove_precursor_peak", "<num>", 0, "0=no, 1=yes, 2=all charge reduced precursor peaks (for ETD)", false, true);
+    registerIntOption_("remove_precursor_tolerance", "<num>", 1.5, "+- Da tolerance for precursor removal", false, true);
+    registerStringOption_("clear_mz_range", "[min] [max]", "0.0 0.0", "for iTRAQ/TMT type data; will clear out all peaks in the specified m/z range", false, true);
```

Comment:
Again, OpenMS generally uses the notation "[min]:[max]" for ranges. See `TOPPBase::parseRange_`.